sdk/tables/Azure.Data.Tables/src/TableEntity.cs

```diff
@@ -235,5 +236,56 @@ private static void EnforceType(Type requestedType, Type givenType)
                     $"Cannot return {requestedType} type for a {givenType} typed property."));
             }
         }
+
+        /// <summary>
+        /// Performs type coercion for numeric types.
+        /// Integers will be coerced to long or double.
+        /// Double and long values will accept integers values but remain double or long.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">The type coercion is not allowed.</exception>
+        private static object CoerceType(object existingValue, object newValue)
+        {
+            if (!existingValue.GetType().IsAssignableFrom(newValue.GetType()))
+            {
+                object convertedValue = existingValue switch
+                {
+                    int _ => newValue switch
+                    {
+                        // All int values can be doubles
+                        double newDoubleValue =>  newDoubleValue,
+                        // All int values can be longs
+                        long newLongValue => newLongValue,
+                        _ => null
+                    },
+                    double _ => newValue switch
+                    {
+                        // if we already had a double value, preserve it as double even if neValue was an int.
+                        // example: entity["someDoubleValue"] = 5;
+                        int newIntValue => (double)newIntValue,
```

Comment:
If `newValue` is already a double, wouldn't this null it out?   I think we may be missing a case, but I'm not sure if that was intended.

Reply:
Discussed offline - this is not an issue since we'll not enter this block if the new and existing values are type assignable
` if (!existingValue.GetType().IsAssignableFrom(newValue.GetType()))